src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -360,16 +380,27 @@ public async Task<uint256> SendManyAsync(string fromAccount, string addressesJso
             }
         }
 
+        /// <summary>
+        /// Will attempt to load the "default" wallet if it exists, else it will simply get the first available.
+        /// </summary>
+        /// <returns>Reference to the default wallet, or the first available.</returns>
         private WalletAccountReference GetAccount()
         {
-            //TODO: Support multi wallet like core by mapping passed RPC credentials to a wallet/account
-            string walletName = this.walletManager.GetWalletsNames().FirstOrDefault();
-            if (walletName == null)
-                throw new RPCServerException(RPCErrorCode.RPC_INVALID_REQUEST, "No wallet found");
-
-            HdAccount account = this.walletManager.GetAccounts(walletName).FirstOrDefault();
+            if (this.walletSettings.DefaultWallet)
+            {
+                HdAccount account = this.walletManager.GetAccounts(WalletManager.DefaultWalletName).Single();
+                return new WalletAccountReference(WalletManager.DefaultWalletName, account.Name);
+            }
+            else
+            {
+                //TODO: Support multi wallet like core by mapping passed RPC credentials to a wallet/account
```

Comment:
I'm not sure what this comment was originally getting at. I think bitcoin core only has a single wallet? If we want to abandon the idea of multiple wallets through RPC completely (I can't see how RPC supports multiple wallets on core), then you may be able to simplify by just first checking if "Default" is there and if not then grabbing the first and then you don't have to have that dependency to WalletSettings.

Accounts is a whole other mess, as seems like bitcoin core has changed the way that works in more recent versions.

Reply:
This is an existing comment, I have not modified it. I agree it is a bit messy, and I seriously dislike the existing implementation of grabbing the first and best wallet (which it does if no default available).

Reply:
I'm happy with getting the default one first. Grabbing the first is a bit random. How is the first even chosen? is it by alphabetical order or something else? The user doesn't know.